src/Arcus.Testing.Logging/SpyLogger.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.Testing.Logging
+{
+    /// <summary>
+    /// Spy (stub) <see cref="ILogger"/> implementation to track the logged messages.
+    /// </summary>
+    public class SpyLogger : ILogger
+    {
+        private readonly ConcurrentBag<LogEntry> _entries = new ConcurrentBag<LogEntry>();
+
+        /// <summary>
+        /// Gets the current logged messages.
+        /// </summary>
+        public IEnumerable<string> Messages => _entries.Select(e => e.Message).Reverse();
```

Comment:
Added `.Reverse()` bc of the way the bag stores things. By adding this, we can have the messages in chronocial order.